### PR TITLE
Force travis to build with clang on MacOS

### PR DESCRIPTION
**Pull request by yiwu-arbug**
Summary:
Attempt to force travis to build with clang on MacOS

Test Plan:
Check the travis job on this PR.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -4,6 +4,9 @@ language: cpp
 os:
   - linux
   - osx
+compiler:
+  - clang
+  - gcc
 osx_image: xcode8.3
```
Comment (sagar0) [inline .travis.yml]: It seems like XCode 8.3 is being used as part of the name for linux builds as well. I still can't figure out why travis is doing so.

**Comment by tamird** [comment]
What is this fixing? How can you tell that gcc is being used on mac?

**Comment by yiwu-arbug** [comment]
@tamird you can see g++ is being use from the console log, take https://travis-ci.org/facebook/rocksdb/jobs/239698584 for example.

**Comment by sagar0** [comment]
@yiwu-arbug that's still clang though, right? g++ and c++ are linked to clang on OS X. 
From the console log:
```
$ g++ --version
Configured with: --prefix=/Applications/Xcode.app/Contents/Developer/usr --with-gxx-include-dir=/usr/include/c++/4.2.1
Apple LLVM version 8.1.0 (clang-802.0.42)
Target: x86_64-apple-darwin16.5.0
Thread model: posix
InstalledDir: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin
```

**Comment by yiwu-arbug** [comment]
@sagar0 Seems you are right.. Just check a previous job and it complains about db_write_test error, which is a signal of it is actual clang: https://travis-ci.org/facebook/rocksdb/jobs/238877014
Should I revert?

**Comment by sagar0** [comment]
I spent some more time investigating this issue, and it seems l like travis by default builds with clang on OS X even when the compiler is specified as gcc. (Ref: [1], [2] and [3]). GCC on OS X is really just a GCC frontend to LLVM. So, this PR is sort of a no-op. 
(If we really want GCC on OS X, then we need to do something like [this](https://github.com/facebook/hhvm/pull/4462/files#diff-308aec47278f474d7aed9791875d4f0bR118).)

[1]: https://github.com/travis-ci/travis-ci/issues/2423
[2]: https://github.com/travis-ci/travis-ci/issues/4587
[3]: https://docs.travis-ci.com/user/languages/cpp/#GCC-on-OS-X

**Comment by siying** [comment]
@sagar0 but why it never caught the CLANG warning we saw internally?

**Comment by yiwu-arbug** [comment]
@siying if you look at https://travis-ci.org/facebook/rocksdb/jobs/238877014 the compiler actual complain about db_write_test, but the job was able to proceed to run the tests. Could there be something wrong with the makefile in non-parallel (i.e. V=1) mode?

**Comment by sagar0** [comment]
@siying I am not sure about that ... may be a different version of clang?
@yiwu-arbug also mentioned that this PR didn't change a few other warnings that he was seeing earlier. So I am pretty sure this PR is a no-op, and we are using clang before and after this PR on OS X.

**Comment by siying** [comment]
@yiwu-arbug I've no idea. Maybe the compiler option was passed in is different?

**Comment by yiwu-arbug** [comment]
Ah, I think I knows why. In the platform_dependent job, the build command is run test command is separated by a semicolon, which makes failure of the build command get ignore.
https://github.com/facebook/rocksdb/blob/master/.travis.yml#L60
I'll move the `make all` command to a separate job.

**Comment by tamird** [comment]
Could you replace the semicolon with '&&' instead? Separate job will
increase build time.

**Comment by siying** [comment]
@yiwu-arbug awesome!

**Pull request closed without merging**